addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRuleUnitTest.java

```diff
@@ -196,12 +197,12 @@ void shouldAlertOnWildcardFrameAncestorsDirective() {
                 alertsRaised.get(0).getOtherInfo(),
                 equalTo(
                         "The following directives either allow wildcard sources (or ancestors), are not "
-                                + "defined, or are overly broadly defined: \nframe-ancestors"
-                                + "\n\nThe directive(s): frame-ancestors are among the directives that do not "
+                                + "defined, or are overly broadly defined: \n" + directive
+                                + "\n\nThe directive(s): " + directive + " are among the directives that do not "
```

Comment:
So I guess I failed and making this as successful test, because I would of course also need to add the other two directives…

Also I guess a test for a missing (not only wildcard) directive is missing here…
No idea how to add that though… :upside_down_face:

Reply:
> Also I guess a test for a missing (not only wildcard) directive is missing here…
No idea how to add that though… 🙃

If they aren't defined and they are among those that don't fallback to default-src then they should already be being alerted upon as wildcard sources.

Reply:
Here's a patch that should do what you need (on top of the current state of this PR at commit [cc7f062](https://github.com/zaproxy/zap-extensions/pull/3627/commits/cc7f0620fe05b79ce71afb156125bce007f7e78c))
```diff
diff --git a/addOns/pscanrules/CHANGELOG.md b/addOns/pscanrules/CHANGELOG.md
index f6581a2..3e2fef1 100644
--- a/addOns/pscanrules/CHANGELOG.md
+++ b/addOns/pscanrules/CHANGELOG.md
@@ -14,6 +14,9 @@
 - Maintenance changes.
 - The CSRF Token scan rule will now raise alerts as Medium risk and Low confidence (Issue 7021).
 
+### Fixed
+- A typo was corrected in the CSP scan rule which was causing invalid assessment of "connect-src" directives.
+
 ## [38] - 2022-01-07
 ### Changed
 - Update minimum ZAP version to 2.11.1.
diff --git a/addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRuleUnitTest.java b/addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRuleUnitTest.java
index 38d8873..61f1e7c 100644
--- a/addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRuleUnitTest.java
+++ b/addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRuleUnitTest.java
@@ -182,11 +182,10 @@
     }
 
     @Test
-    @ValueSource(strings = {"frame-ancestors", "form-action", "connect-src"})
-    void shouldAlertOnWildcardFrameAncestorsDirective(String directive) {
+    void shouldAlertOnWildcardFrameAncestorsDirective() {
         // Given
         HttpMessage msg =
-                createHttpMessage(directive + " *; default-src 'self'; form-action 'none'");
+                createHttpMessage("frame-ancestors *; default-src 'self'; form-action 'none'");
         // When
         scanHttpResponseReceive(msg);
         // Then
@@ -197,12 +196,38 @@
                 alertsRaised.get(0).getOtherInfo(),
                 equalTo(
                         "The following directives either allow wildcard sources (or ancestors), are not "
-                                + "defined, or are overly broadly defined: \n" + directive
-                                + "\n\nThe directive(s): " + directive + " are among the directives that do not "
+                                + "defined, or are overly broadly defined: \nframe-ancestors"
+                                + "\n\nThe directive(s): frame-ancestors are among the directives that do not "
                                 + "fallback to default-src, missing/excluding them is the same as allowing anything."));
         assertThat(
                 alertsRaised.get(0).getEvidence(),
-                equalTo(directive + " *; default-src 'self'; form-action 'none'"));
+                equalTo("frame-ancestors *; default-src 'self'; form-action 'none'"));
+        assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_MEDIUM));
+        assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
+        assertThat(alertsRaised.get(0).getAlertRef(), equalTo("10055-4"));
+    }
+
+    @Test
+    void shouldAlertOnWildcardConnectSourceDirective() {
+        // Given
+        HttpMessage msg =
+                createHttpMessage("connect-src *; default-src 'self'; form-action 'none'");
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised.size(), equalTo(1));
+
+        assertThat(alertsRaised.get(0).getName(), equalTo("CSP: Wildcard Directive"));
+        assertThat(
+                alertsRaised.get(0).getOtherInfo(),
+                equalTo(
+                        "The following directives either allow wildcard sources (or ancestors), are not "
+                                + "defined, or are overly broadly defined: \nconnect-src, frame-ancestors"
+                                + "\n\nThe directive(s): frame-ancestors are among the directives that do not "
+                                + "fallback to default-src, missing/excluding them is the same as allowing anything."));
+        assertThat(
+                alertsRaised.get(0).getEvidence(),
+                equalTo("connect-src *; default-src 'self'; form-action 'none'"));
         assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_MEDIUM));
         assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
         assertThat(alertsRaised.get(0).getAlertRef(), equalTo("10055-4"));
```

Reply:
Okay split your diff up in some comits for clean commits (you can squash it later anyway if you want) and comittet it… thanks a lot… :astonished:

Reply:
> Just say the word, I'm happy to if that helps.

Naw, wanted to do it myself now… :upside_down_face:

Reply:
Also having fun bringing git to properly recognize the co author and my signoff message now… :laughing:

Reply:
Npote: next time I would have rolled back my bad commit and restartet from scratch – that's why you do seperate commits after all… however, as I had your diff we have a crazy commit in the history now… :upside_down_face: